src/state.rs

```diff
@@ -341,9 +371,36 @@ impl DemandProxyState {
                     .on_demand_dns_cache_misses
                     .get_or_create(&labels)
                     .inc();
-                // TODO: optimize so that if multiple requests to the same hostname come in at the same time,
+                // optimize so that if multiple requests to the same hostname come in at the same time,
```

Comment:
Is this comment still valid?

Reply:
Yes, I removed the `TODO` to show the optimization purpose. Do you think it's necessary? If no, I can remove this comment.

Reply:
IC